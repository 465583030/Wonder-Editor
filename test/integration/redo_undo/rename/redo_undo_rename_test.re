open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: rename", () => {
    let sandbox = getSandboxDefaultVal();
    beforeEach(() => {
      sandbox := createSandbox();
      StateHistoryToolEditor.clearAllState();
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));
    describe("test simulate set currentSceneTreeNode", () => {
      beforeEach(() => {
        TestTool.closeContractCheck();
        MainEditorSceneTool.initStateAndGl(~sandbox, ());
        MainEditorSceneTool.createDefaultScene(
          sandbox,
          MainEditorSceneTool.setFirstBoxTobeCurrentSceneTreeNode,
        );
        SceneTreeTool.setSceenTreeSpecificGameObject(1);
      });
      afterEach(() => TestTool.openContractCheck());

      describe("test undo one step", () =>
        test("step which from second to first", () => {
          let currentGameObjectTransform =
            GameObjectTool.getCurrentSceneTreeNodeTransform();
          TransformEventTool.simulateTwiceChangeEvent(
            currentGameObjectTransform,
          );
          StateHistoryToolEditor.undo();
          BuildComponentTool.buildMainEditorTransformComponent(
            TestTool.buildEmptyAppState(),
            currentGameObjectTransform,
          )
          |> ReactTestTool.createSnapshotAndMatch;
        })
      );
    });
  });