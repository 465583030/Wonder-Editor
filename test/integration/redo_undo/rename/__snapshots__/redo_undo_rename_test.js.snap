// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: rename test simulate set currentSceneTreeNode test undo one step step which from second to first 1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="wonder-float-input"
  >
    <span
      className="component-label"
    >
      X : 
    </span>
    <input
      className="input-component float-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="11.25"
    />
  </article>
  <article
    className="wonder-float-input"
  >
    <span
      className="component-label"
    >
      Y : 
    </span>
    <input
      className="input-component float-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="0"
    />
  </article>
  <article
    className="wonder-float-input"
  >
    <span
      className="component-label"
    >
      Z : 
    </span>
    <input
      className="input-component float-input"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="0"
    />
  </article>
</article>
`;
